cupy/core/core.pyx

```diff
@@ -2118,6 +2126,63 @@ cdef _argmax = create_reduction_func(
     None, _min_max_preamble)
 
 
+cdef _round_preamble = '''
+template<typename T> __device__ T pow10(long long n){
+  T x = 1, a = 10;
+  while (n) {
+    if (n & 1) x *= a;
+    a *= a;
+    n >>= 1;
+  }
+  return x;
+};
+'''
+
+
+cdef _round_float = '''
+if (in1 == 0) {
+    out0 = round(in0);
+} else {
+    double x = pow10<double>(abs(in1));
```

Comment:
Should not calculate this in the loop body.

Reply:
I added